src/schema/types.rs

```diff
@@ -718,15 +718,26 @@ impl SchemaDescriptor {
 
   /// Returns column root [`Type`](`::schema::types::Type`) for a field position.
   pub fn get_column_root(&self, i: usize) -> &Type {
+    let result = self.column_root_of(i);
+    assert!(result.is_some(), "Expected a value for index {} but found None", i);
+    result.unwrap().as_ref()
+  }
+
+  /// Returns column root [`Type`](`::schema::types::Type`) pointer for a field position.
+  pub fn get_column_root_ptr(&self, i: usize) -> TypePtr {
```

Comment:
I think it would be good to move the assertion into `column_root_of` method and return either Rc directly or ref extracted from Rc. In this case we do not need to duplicate the error message.